clouddriver-security/src/test/groovy/com/netflix/spinnaker/clouddriver/security/DefaultAllowedAccountsValidatorSpec.groovy

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.security
+
+import com.netflix.spinnaker.clouddriver.security.AccountCredentials
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider
+import com.netflix.spinnaker.clouddriver.security.DefaultAllowedAccountsValidator
+import org.springframework.validation.Errors
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class DefaultAllowedAccountsValidatorSpec extends Specification {
+  TestAccountCredentials credentialsWithRequiredGroup = new TestAccountCredentials(name: 'TestAccount', requiredGroupMembership: ['targetAccount1'])
+
+  @Unroll
+  void "should reject if allowed accounts does not intersect with required group memberships"() {
+    given:
+    def errors = Mock(Errors)
+    def validator = new DefaultAllowedAccountsValidator(Mock(AccountCredentialsProvider) {
+      1 * getAll() >> { [credentialsWithRequiredGroup] }
+      0 * _
+    })
+
+    when:
+    def description = new TestDescription(credentials: credentialsWithRequiredGroup)
+
+    validator.validate("TestUser", allowedAccounts, description, errors)
+
+    then:
+    rejectValueCount * errors.rejectValue("credentials", "unauthorized", _)
+
+    where:
+    allowedAccounts                      || rejectValueCount
```

Comment:
It's legit -https://spockframework.github.io/spock/docs/1.0/data_driven_testing.html#_syntactic_variations